src/main/java/com/mraof/minestuck/block/machine/IntellibeamLaserstationBlock.java

```diff
@@ -0,0 +1,114 @@
+package com.mraof.minestuck.block.machine;
+
+import com.mraof.minestuck.block.BlockUtil;
+import com.mraof.minestuck.block.MSProperties;
+import com.mraof.minestuck.blockentity.HolopadBlockEntity;
+import com.mraof.minestuck.blockentity.IntellibeamLaserstationBlockEntity;
+import com.mraof.minestuck.blockentity.MSBlockEntityTypes;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.EntityBlock;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.entity.BlockEntityTicker;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+public class IntellibeamLaserstationBlock extends MachineBlock implements EntityBlock
+{
+	public static final Map<Direction, VoxelShape> SHAPE = createRotatedShapes(2, 0, 1, 14, 6, 13);
+	public static final Map<Direction, VoxelShape> COLLISION_SHAPE;
+	public static final BooleanProperty HAS_CARD = MSProperties.HAS_CARD;
+	
+	static
+	{
+		VoxelShape topShape = Block.box(3, 6, 3, 13, 7, 13);
+		COLLISION_SHAPE = createRotatedShapes(4, 0, 14, 12, 10, 16);
+		COLLISION_SHAPE.replaceAll((enumFacing, shape) -> Shapes.or(shape, topShape));
+	}
+	
+	public IntellibeamLaserstationBlock(BlockBehaviour.Properties builder)
+	{
+		super(builder);
+		registerDefaultState(defaultBlockState().setValue(HAS_CARD, false));
+	}
+	
+	@Nullable
+	@Override
+	public BlockEntity newBlockEntity(BlockPos pos, BlockState state)
+	{
+		return new HolopadBlockEntity(pos, state);
```

Comment:
switch to using IntellibeamLaserstationBlockEntity

Reply:
done